app/controllers/staff/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Staff::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def azure_activedirectory_v2
+    auth = request.env["omniauth.auth"]
+    email = auth["info"]["email"]
+    provider = auth["provider"]
+    azure_ad_uid = auth["azure_ad_uid"]
+
+    @staff = Staff.find_by_email(email)
+
+    if @staff&.update(provider:, azure_ad_uid:)
```

Comment:
```suggestion
    if @staff&.update(azure_ad_uid:)
```

I think this will always fail as the `provider` is no longer there.